src/Root.tsx

```diff
@@ -17,13 +20,17 @@ const Root: FC = () => {
 
   return (
     <Web3ContextProvider>
-      <Provider store={store}>
-        <I18nProvider i18n={i18n}>
-          <BrowserRouter basename={"/#"}>
-            <App />
-          </BrowserRouter>
-        </I18nProvider>
-      </Provider>
+      <NewWeb3ContextProvider>
+        <QueryClientProvider client={queryClient}>
```

Comment:
can you explain the logic in moving QueryClientProvider up here?

Reply:
Just because fewer re-renders occur in the Root component compared to the App component at the moment, and re-rendering the react-query provider is pointless.

Reply:
nice, thank you ser